### PR TITLE
refactor (bbb-html5): talking indicator migration to graphql

### PR DESCRIPTION
**Pull request by Scroody**
### What does this PR do?

This PR refactors the talking indicator component to type-script and graphql.

### CODE REVIEW

**Review by gustavotrott — APPROVED** [review]
Working well! Good job @Scroody !

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/styles.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
+import styled from 'styled-components';
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/styles.js:1]: both style and service aren't using the right suffix

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+} from "../queries";
+import Service from '../talking-indicator/service';
+import Styled from './styles';
+import { Meteor } from 'meteor/meteor';
+import SpeechService from '/imports/ui/components/audio/captions/speech/service';
+import { debounce } from "radash";
+import { makeCall } from "/imports/ui/services/api";
+import { User } from "/imports/ui/Types/user";
+import { uniqueId } from '/imports/utils/string-utils';
+import { CurrentUser } from "/imports/api/users";
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:16]: You should not use meteor data in the component

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+import React from "react";
+import { defineMessages, useIntl } from 'react-intl';
+import {
+    TALKING_INDICATOR_SUBSCRIPTION,
+    MEETING_IS_BREAKOUT,
+} from "../queries";
+import Service from '../talking-indicator/service';
+import Styled from './styles';
+import { Meteor } from 'meteor/meteor';
+import SpeechService from '/imports/ui/components/audio/captions/speech/service';
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:11]: You should isolate the component and remove this kind of import

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    moreThanMaxIndicatorsWereTalking: {
+        id: 'app.talkingIndicator.moreThanMaxIndicatorsWereTalking',
+        description: 'indicator label for all users who is not talking but not visible',
+    },
+});
+
+const TALKING_INDICATORS_MAX = 8;
+const TALKING_INDICATOR_MUTE_INTERVAL = 500;
+
+const isBreakoutRoom = MEETING_IS_BREAKOUT;
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:48]: What are you doing here?

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+const PUBLIC_CONFIG = Meteor.settings.public;
+const ROLE_MODERATOR = PUBLIC_CONFIG.user.role_moderator;
+const amIModerator = CurrentUser.role === ROLE_MODERATOR;
+
+interface TalkingIndicatorProps {
+    muteUser: Function,
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:54]: You are declaring more props that you use

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+
+        for (let i = 0; i < maxNumberVoiceUsersNotification; i += 1) {
+            const name = usersTalking[i].user.name;
+            const userId = usersTalking[i].user.userId;
+            const color = usersTalking[i].user.color;
+            const {
+                talking, floor, muted,
+            } = usersTalking[i];
+
+            talkers[`${userId}`] = {
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:204]: Why to you convert it on a objecto to after take its keys and use as an array?

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+                        muted={muted}
+                        talking={talking}
+                    />
+                )}
+                <Styled.TalkingIndicatorButton
+                    $spoke={!talking || undefined}
+                    $muted={muted}
+                    $isViewer={!amIModerator || undefined}
+                    key={uniqueId(`${name}-`)}
+                    onClick={() => this.handleMuteUser(id)}
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:110]: there is no declaration for this function

**Review by Tainan404 — APPROVED** [review]

**Review by Tainan404 — CHANGES_REQUESTED** [review]
.

File: bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx [hunk 8]
```diff
@@ -45,9 +44,7 @@ const TALKING_INDICATORS_MAX = 8;
 const TALKING_INDICATOR_MUTE_INTERVAL = 500;
 
 const isBreakoutRoom = MEETING_IS_BREAKOUT;
-const PUBLIC_CONFIG = Meteor.settings.public;
-const ROLE_MODERATOR = PUBLIC_CONFIG.user.role_moderator;
-const amIModerator = CurrentUser.role === ROLE_MODERATOR;
+const amIModerator = AM_I_MODERATOR;
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/nav-bar/talking-indicator-graphql/component.tsx:47]: https://www.apollographql.com/docs/react/data/subscriptions/#executing-a-subscription

File: bigbluebutton-html5/imports/ui/components/nav-bar/queries.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  }`;
+
+export const MEETING_IS_BREAKOUT = gql ` subscription MeetingIsBreakout {
+    meeting {
+      isBreakout
+    }
+  }`;
+
+  export const AM_I_MODERATOR = gql ` subscription AmIModerator{
+    user {
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/components/nav-bar/queries.ts:26]: Request user info from endpoint `user_current` instead of `user`.

Comment (Scroody) [reply bigbluebutton-html5/imports/ui/components/nav-bar/queries.ts]: Done!

**Pull request closed without merging**